python/samples/agentchat_graphrag/app.py

```diff
@@ -0,0 +1,77 @@
+import argparse
+import asyncio
+import json
+import logging
+from typing import Any, Dict
+from autogen_agentchat.messages import TextMessage
+from autogen_ext.tools.graphrag import (
+    GlobalSearchTool,
+    LocalSearchTool,
+)
+from autogen_agentchat.agents import AssistantAgent
+from autogen_agentchat.conditions import TextMentionTermination, MaxMessageTermination
+from autogen_agentchat.teams import RoundRobinGroupChat
+from autogen_core.models import ChatCompletionClient
+
+
+async def main(model_config: Dict[str, Any]) -> None:
+    # Initialize the model client from config
+    model_client = ChatCompletionClient.load_component(model_config)
+
+    # Set up global search tool
+    global_tool = GlobalSearchTool.from_settings(
+        settings_path="./settings.yaml"
+    )
+
+    local_tool = LocalSearchTool.from_settings(
+        settings_path="./settings.yaml"
+    )
+
+    # Create assistant agent with both search tools
+    assistant_agent = AssistantAgent(
+        name="search_assistant",
+        tools=[global_tool, local_tool],
+        model_client=model_client,
+        system_message=(
+            "You are a tool selector AI assistant using the GraphRAG framework. "
+            "Your primary task is to determine the appropriate search tool to call based on the user's query. "
+            "For specific, detailed information about particular entities or relationships, call the 'local_search' function. "
+            "For broader, abstract questions requiring a comprehensive understanding of the dataset, call the 'global_search' function. "
+            "Do not attempt to answer the query directly; focus solely on selecting and calling the correct function."
+        )
+    )
+
+    # Set up the team
+    termination = TextMentionTermination("TERMINATE") | MaxMessageTermination(10)
+    team = RoundRobinGroupChat(
+        participants=[assistant_agent],
+        termination_condition=termination
+    )
+
+    # Run a sample query
+    query = "What does the station-master says about Dr. Becher?"
+    print(f"\nQuery: {query}")
+    
+    response_stream = team.run_stream(task=query)
+    async for msg in response_stream:
+        if isinstance(msg, TextMessage) and hasattr(msg, "content"):
+            print(f"\nAgent response: {msg.content}")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Run a GraphRAG search with an agent.")
+    parser.add_argument("--verbose", action="store_true", help="Enable verbose logging.")
+    # See agentchat_chainlit/model_config_template.json for model_config.json
```

Comment:
Let's make the example self-contained so let's add the `model_config_template.json` file and explain how to set up model configs in README.md